### PR TITLE
move configuration into a concern

### PR DESCRIPTION
**Pull request by tilsammans**
This makes it obvious where to document this module.
And it DRYs things up.

### CODE REVIEW

File: app/models/pageflow/chapter.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
     belongs_to :storyline, touch: true
     has_many :pages, -> { order('position ASC') }
 
     delegate :entry, to: :storyline
 
-    serialize :configuration, JSON
-
     def configuration
       super || {}
     end
```
Comment (tf) [inline app/models/pageflow/chapter.rb]: Shouldn't this be removed as well?

**Comment by tilsammans** [comment]
I'll clean it up.

I should have realised that configuration was taken. Ahhh, naming things...

**Comment by tilsammans** [comment]
How about `Pageflow::Attributes` or a variation thereof? `Pageflow::PageAttributes` perhaps?

**Comment by tf** [comment]
`PageAttributes` is confusing since the mixin is also used for storylines and chapters. `Pageflow::Attributes`  sounds a bit too unspecific. How about `Pageflow::ConfigurationAttribute`?

**Comment by tilsammans** [comment]
One more: since it's basically a key/value or object store, is `Pageflow::ConfigurationStore` or `Pageflow::ConfigurationStorage` a good name?

**Comment by tf** [comment]
Technically, though, the page/chapter/storyline that includes the mixin is not really the key/value store itself, but rather provides an attribute to access the key/value store. Maybe even more specific, something like `WithSerializedConfigruation`. But maybe that's too technical.

**Comment by tilsammans** [comment]
Waiting for #803 to continue.

Maybe just `Pageflow::SerializedConfiguration` as a concern name? It's descriptive at least. We are still sort of abusing configuration to store headers and such, but that is for another time.

**Comment by tf** [comment]
Since #694 `Pageflow::Widget` now also has a serialzed configuration and could therefore also use the mixin.

**Comment by tilsammans** [comment]
I implemented the changes.

**Pull request merged**